handlers/messagecontext_test.go

```diff
@@ -0,0 +1,42 @@
+package handlers_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	servicebus "github.com/Azure/azure-service-bus-go"
+	"github.com/Azure/go-shuttle/handlers"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_CancelContextWhenDoneWithMessage(t *testing.T) {
+	next := &FakeHandler{}
+	handler := handlers.NewDeadlineContext(next)
+	ctx := context.Background()
+	ttlMessage := &servicebus.Message{}
+	err := handler.Handle(ctx, ttlMessage)
+	assert.Nil(t, err)
+	assert.Equal(t, context.Canceled, next.Ctx.Err())
+}
+
+func Test_CancelContextWhenDeadlineExceeded(t *testing.T) {
```

Comment:
This actually tests what was already happening. If we want to test the bug it would have to be cancel is called when deadline is not exceeded.